From df4dc12513435bcfb03707c4c41974d98afe8d14 Mon Sep 17 00:00:00 2001
From: leocheng-tw <Leo_Cheng@phoenix.com>
Date: Wed, 9 Feb 2022 08:26:06 +0000
Subject: [PATCH] Correct Redfish URL for BC and remove "Power cap setting"
 checkbox. User need to enable power cap by applying IPMI command.

---
 .../ResourceManagement/PowerControlStore.js       |  4 ++--
 src/views/ResourceManagement/Power.vue            | 15 ++++++++-------
 2 files changed, 10 insertions(+), 9 deletions(-)
 mode change 100644 => 100755 src/store/modules/ResourceManagement/PowerControlStore.js
 mode change 100644 => 100755 src/views/ResourceManagement/Power.vue

diff --git a/src/store/modules/ResourceManagement/PowerControlStore.js b/src/store/modules/ResourceManagement/PowerControlStore.js
old mode 100644
new mode 100755
index 9dbddf0..8d4a00e
--- a/src/store/modules/ResourceManagement/PowerControlStore.js
+++ b/src/store/modules/ResourceManagement/PowerControlStore.js
@@ -23,7 +23,7 @@ const PowerControlStore = {
     },
     async getPowerControl({ commit }) {
       return await api
-        .get('/redfish/v1/Chassis/chassis/Power')
+        .get('/redfish/v1/Chassis/BC_Baseboard/Power')
         .then((response) => {
           const powerControl = response.data.PowerControl;
           const powerCap = powerControl[0].PowerLimit.LimitInWatts;
@@ -43,7 +43,7 @@ const PowerControlStore = {
       };
 
       return await api
-        .patch('/redfish/v1/Chassis/chassis/Power', data)
+        .patch('/redfish/v1/Chassis/BC_Baseboard/Power', data)
         .then(() =>
           i18n.t('pageServerPowerOperations.toast.successSaveSettings')
         )
diff --git a/src/views/ResourceManagement/Power.vue b/src/views/ResourceManagement/Power.vue
old mode 100644
new mode 100755
index 8e4acdd..baade41
--- a/src/views/ResourceManagement/Power.vue
+++ b/src/views/ResourceManagement/Power.vue
@@ -22,13 +22,13 @@
         <b-row>
           <b-col sm="8" md="6" xl="12">
             <b-form-group :label="$t('pagePower.powerCapSettingLabel')">
-              <b-form-checkbox
-                v-model="isPowerCapFieldEnabled"
-                data-test-id="power-checkbox-togglePowerCapField"
-                name="power-cap-setting"
-              >
-                {{ $t('pagePower.powerCapSettingData') }}
-              </b-form-checkbox>
+              <dd>
+                {{
+                  isPowerCapFieldEnabled
+                    ? $t('global.status.enabled')
+                    : $t('global.status.disabled')
+                }}
+              </dd>
             </b-form-group>
           </b-col>
         </b-row>
@@ -72,6 +72,7 @@
         </b-row>
 
         <b-button
+          :disabled="!isPowerCapFieldEnabled"
           variant="primary"
           type="submit"
           data-test-id="power-button-savePowerCapValue"
